SimPEG/EM/Static/DC/IODC.py

```diff
@@ -510,9 +510,26 @@ def set_mesh(self, topo=None,
                 )
             x0 = self.electrode_locations[:, 0].min()
             if topo is None:
-                locs = np.sort(self.electrode_locations, axis=0)
+                # For 2D mesh
+                if dimension == 2:
+                    # sort by x
+                    row_idx = np.lexsort((self.electrode_locations[:, 0],))
+                # For 3D mesh
+                else:
+                    # sort by x, then by y
+                    row_idx = np.lexsort((self.electrode_locations[:, 1],
+                                          self.electrode_locations[:, 0]))
+                locs = self.electrode_locations[row_idx, :]
             else:
-                locs = np.vstack((topo, self.electrode_locations))
```

Comment:
my thought process here was including both electrode locations and topo locations. 
so we we still need `locs_tmp = np.vstack(np.vstack((topo, self.electrode_locations))`
then line 532 need to be changed as 

`locs = loc_tmp[row_index, :]`

Reply:
My consideration is that when the topography and electrode are at the same x (or x, y) coordinates (such as the following settings), surface2ind_topo may have the wrong interpolation.

```
# topo and electrode are at the same x coordinate, but z is different
topo = np.array([[0, 10], [5, 2]])
electrode_locations = np.array([[0, 0], [5, 3]]) 
```

https://github.com/simpeg/simpeg/blob/46c6619c3368dd0000c3b2fe27bdbd042fe8c17a/SimPEG/EM/Static/DC/IODC.py#L592

Should we just keep the topo location and add a warning or just throw an error?